cmd/image-builder/manifest.go

```diff
@@ -30,29 +30,15 @@ func depsolve(cacheDir string, packageSets map[string][]rpmmd.PackageSet, d dist
 }
 
 func outputManifest(out io.Writer, distroName, imgTypeStr, archStr string) error {
-	filterExprs := []string{
-		fmt.Sprintf("name:%s", distroName),
-		fmt.Sprintf("arch:%s", archStr),
-		fmt.Sprintf("type:%s", imgTypeStr),
-	}
-	filteredResults, err := getFilteredImages(filterExprs)
+	// XXX: what/how much do we expose here?
+	var options distro.ImageOptions
```

Comment:
For on-prem, we use `ImageOptions` for the stuff that is passed on the CLI and is not part of the BP. E.g. image size.

Reply:
This is something we need to think about.  The image options, like Tomáš said, is (mostly) for things that come from the command line.  This includes:
- image size
- ostree options (url, ref, and parent, when using `composer-cli compose start-ostree`).

https://github.com/osbuild/osbuild-composer/blob/a7cd521325504a43d3708f8f91ab63999862b279/internal/weldr/api.go#L2549-L2556

But it's also used for other things that get set automatically (or semi-automatically):
- Facts: this shouldn't be exposed at all.  It adds a value to the RHSM facts so we can see how the image was built when it registers back with subscription manager (hosted service vs on-prem).  Perhaps we could add a new value for this now for the daemonless builder.
- Subscription: Used only by the service (I think, https://github.com/osbuild/osbuild-composer/blob/a7cd521325504a43d3708f8f91ab63999862b279/internal/cloudapi/v2/compose.go#L1194).  But might be useful to expose for on-prem builders.
- PartitioningMode: This is a funny one.  In osbuild-composer, it's set in the blueprint (https://github.com/osbuild/osbuild-composer/blob/a7cd521325504a43d3708f8f91ab63999862b279/internal/blueprint/customizations.go#L23), but it's passed down to images through the image options (https://github.com/osbuild/osbuild-composer/blob/a7cd521325504a43d3708f8f91ab63999862b279/internal/weldr/api.go#L2552).  Not sure what we want to do there.  Since it's in the BP in composer already, I think we should move it there in images too.